RoddyCore/test/src/de/dkfz/roddy/execution/io/LocalExecutionServiceSpec.groovy

```diff
@@ -16,52 +16,55 @@ class LocalExecutionServiceSpec extends Specification {
 
     LocalExecutionService es = ExecutionService.instance
 
-    def "execute('echo')"() {
+    def "execute synchronously and succeed with captured stdout and non!-ignored stderr"() {
         when:
-        ExecutionResult res = es.execute("echo 'hello'", true)
+        ExecutionResult res = es.execute("echo 'hello'; echo 'error' >> /dev/stderr", true)
 
         then:
         res.exitCode == 0
         res.successful
-        res.resultLines.size() == 1
+        res.resultLines.size() == 2
         res.resultLines[0] == "hello"  // Note that the newline is removed.
+        res.resultLines[1] == "error"
     }
 
-    def "execute('echo \'hello\'; false')"() {
+    def "execute synchronously and fail with captured stdout and non!-ignored stderr"() {
         when:
-        ExecutionResult res = es.execute("echo 'hello'; false", true)
+        ExecutionResult res = es.execute("echo 'hello'; echo 'error' >> /dev/stderr; false", true)
 
         then:
         res.exitCode == 1
         !res.successful
-        res.resultLines.size() == 1
+        res.resultLines.size() == 2
         res.resultLines[0] == "hello"  // Note that the newline is removed.
+        res.resultLines[1] == "error"
     }
 
-    def "execute('echo \"hello\"; false', waitFor = false)"() {
+    def "execute synchronously and succeed with empty stdout and non!-ignored stderr"() {
```

Comment:
shouldn't that be a "execute synchronously and *fail*" (exit code = 1)?

Reply:
Sorry. Maybe a tabular test would have been more appropriate for all these parameter combinations and results.